OpenRA.Mods.D2k/Activities/SwallowActor.cs

```diff
@@ -89,21 +89,26 @@ bool WormAttack(Actor worm)
 			}
 
 			positionable.SetPosition(worm, targetLocation);
+			PlayAttack(worm);
 			foreach (var notify in worm.TraitsImplementing<INotifyAttack>())
 				notify.Attacking(worm, target, null, null);
-			PlayAttackAnimation(worm);
 
 			var attackPosition = worm.CenterPosition;
 			var affectedPlayers = lunch.Select(x => x.Owner).Distinct();
-			foreach (var affectedPlayer in affectedPlayers)
-				NotifyPlayer(affectedPlayer, attackPosition);
+
+			worm.World.AddFrameEndTask(_ =>
+			{
+				foreach (var affectedPlayer in affectedPlayers)
+					NotifyPlayer(affectedPlayer, attackPosition);
+			});
 
 			return true;
 		}
 
-		void PlayAttackAnimation(Actor self)
+		void PlayAttack(Actor self)
 		{
 			renderUnit.PlayCustomAnim(self, "mouth");
+			Sound.Play(swallow.Info.WormAttackSound, self.CenterPosition);
```

Comment:
A `Sound.PlayThen` that would `NotifyPlayer` afterwards would be useful here for polishing.